includes/emails/class-wc-subscriptions-email-preview.php

```diff
@@ -141,13 +145,46 @@ private function get_dummy_address() {
 		return apply_filters( 'woocommerce_subscriptions_email_preview_dummy_address', $address, $this->email_type );
 	}
 
+	/**
+	 * Creates a dummy retry for use when previewing failed subscription payment retry emails.
+	 *
+	 * @param WC_Order $order The order object to create a dummy retry for.
+	 * @return WCS_Retry The dummy retry object.
+	 */
+	private function get_dummy_retry( $order ) {
+
+		if ( ! class_exists( 'WCS_Retry_Manager' ) || ! is_a( $order, 'WC_Order' ) ) {
+			return null;
+		}
+
+		$retry_number = 1;
+		$retry_rule   = WCS_Retry_Manager::rules()->get_rule( $retry_number, $order->get_id() );
```

Comment:
Very minor but since the set of retry rules is filterable, should we confirm `get_rule()` is not null to avoid calling `$retry_rule->get_retry_interval()` on null type issues.

It would be silly that someone would filter the list of retry rules and set it to an empty list so I'm not sure if it's worth us coding defensively to prevent this but wanted to call it out 😅 

Not even sure what would be displayed if it's null because if we mock a default value if there's no retry rule, then that's not an accurate preview.

---

Snippet to reproduce the issue:

```php
add_filter( 'wcs_default_retry_rules', function( $rules ) {
	return [];
} );
```

Reply:
yeah if some 3rd party code is not returning a valid retry object than our only option is to really falsify some defaults. I decided that's probably the best approach to make the preview still work. 

I've fixed that in https://github.com/Automattic/woocommerce-subscriptions-core/pull/815/commits/54f3f2759a1816f1f30bcd910dae0f3338d86ab5